src/TraceEvent/Parsers/LinuxKernelEventParser.cs

```diff
@@ -0,0 +1,212 @@
+                templates[1] = new ProcessStopTraceData(null, 2, 1, "Process", ProcessTaskGuid, 2, "Stop", ProviderGuid, ProviderName);
+                s_templates = templates;
+            }
+            foreach (var template in s_templates)
+                if (eventsToObserve == null || eventsToObserve(template.ProviderName, template.EventName) == EventFilterResponse.AcceptEvent)
+                    callback(template);
+        }
+
+        #endregion
+    }
+}
+
+namespace Microsoft.Diagnostics.Tracing.Parsers.LinuxKernel
+{
+    public sealed class ProcessStartTraceData : TraceEvent
+    {
+        #region Private
+        internal ProcessStartTraceData(Action<ProcessStartTraceData> action, int eventID, int task, string taskName, Guid taskGuid, int opcode, string opcodeName, Guid providerGuid, string providerName)
+            : base(eventID, task, taskName, taskGuid, opcode, opcodeName, providerGuid, providerName)
+        {
+            Action = action;
+        }
+        protected internal override void Dispatch()
+        {
+            Action(this);
+        }
+        protected internal override void Validate()
+        {
+            //Debug.Assert(!(Version == 0 && EventDataLength != SkipUnicodeString(4)));
+            //Debug.Assert(!(Version > 0 && EventDataLength < SkipUnicodeString(4)));
+        }
+        protected internal override Delegate Target
+        {
+            get { return Action; }
+            set { Action = (Action<ProcessStartTraceData>) value; }
+        }
+        public override StringBuilder ToXml(StringBuilder sb)
+        {
+             Prefix(sb);
+             XmlAttrib(sb, "ProcessID", ProcessID);
+             XmlAttrib(sb, "ProcessName", ProcessName);
+             sb.Append("/>");
+             return sb;
+        }
+
+        public override string[] PayloadNames
+        {
+            get
+            {
+                if (payloadNames == null)
+                    payloadNames = new string[] { "ProcessID", "ProcessName"};
+                return payloadNames;
+            }
+        }
+
+        public override object PayloadValue(int index)
+        {
+            switch (index)
+            {
+                case 0:
+                    return ProcessID;
+                case 1:
+                    return ProcessName;
+                default:
+                    Debug.Assert(false, "Bad field index");
+                    return null;
+            }
+        }
+
+        public static ulong GetKeywords() { return 1; }
+        public static string GetProviderName() { return "Linux-Kernel"; }
+        public static Guid GetProviderGuid() { return new Guid("4da4eb17-cd7d-4b13-b807-22908d2996b5"); }
+        private event Action<ProcessStartTraceData> Action;
+        #endregion
+    }
+    public sealed class ProcessStopTraceData : TraceEvent
+    {
+/*      public int ProcessID { get { return GetInt32At(0); } }
```

Comment:
Can you please add some comments around the commented out sections of code in this file?  I think most things are commented out because you're actually getting the data from the context stream instead of the event payload, which is fine.  I just want it to be clear to readers in the future.  If you think any of the commented-out code should never be present then let's remove it.